src/LuaTransformer.ts

```diff
@@ -4634,6 +4636,76 @@ export class LuaTransformer {
         return this.createSelfIdentifier(thisKeyword);
     }
 
+    public transformTaggedTemplateExpression(expression: ts.TaggedTemplateExpression): ExpressionVisitResult {
+        const strings: ts.StringLiteral[] = [];
+        const rawStrings: string[] = [];
+        const expressions: ts.Expression[] = [];
+        if (ts.isTemplateExpression(expression.template)) {
+            // Expressions are in the string.
+            strings.push(ts.createStringLiteral(expression.template.head.text));
+            rawStrings.push(expression.template.head.getText());
+            strings.push(...expression.template.templateSpans.map(span => ts.createStringLiteral(span.literal.text)));
+            rawStrings.push(...expression.template.templateSpans.map(span => span.literal.getText()));
+            expressions.push(...expression.template.templateSpans.map(span => span.expression));
+        } else if (ts.isNoSubstitutionTemplateLiteral(expression.template)) {
+            // No expressions are in the string.
+            strings.push(ts.createStringLiteral(expression.template.text));
+            rawStrings.push(expression.template.getText());
+        } else {
+            throw TSTLErrors.UnsupportedKind("TaggedTemplateExpression", expression.kind, expression);
+        }
+
+        // Preserve some special characters
+        const preservedRawStrings = rawStrings.map(rawString => {
+            const preservedString = rawString
+                .replace(/`/g, "")
+                .replace(/\$/g, "")
+                .replace(/\{/g, "")
+                .replace(/\}/g, "")
```

Comment:
I'm confused about why these characters are being stripped. Shouldn't the raw version of `` `foo$` `` be `"foo$"`, and `` `foo\${1}` `` be `"foo\\${1}"`?

Reply:
Changed the replacement code and added some more tests for these cases